### PR TITLE
Bug 1440421 - Simple implementation of mirroring for feedback

### PR DESCRIPTION
**Pull request by dmose**
This works together with the other patch in bug 1440421.  That patches push down browser/extensions/activity-stream into browser/extensions/activity-stream/system-addon in a way that still builds and functions.  This one updates the exporting code to export everything.

### CODE REVIEW

File: yamscripts.yml [hunk 1]
```diff
@@ -21,11 +21,10 @@ scripts:
     webpack: webpack --config webpack.system-addon.config.js --display-optimization-bailout
     css: node-sass --source-map true --source-map-contents system-addon/content-src/styles -o system-addon/css
     html: rimraf system-addon/prerendered && webpack --config webpack.prerender.config.js && node ./bin/render-activity-stream-html.js
-    # Copy over all of the system-addon directory
-    copy: cpx "system-addon/**/{,.}*" $npm_package_config_mc_dir/browser/extensions/activity-stream
-    version: node ./bin/update-version.js $npm_package_config_mc_dir/browser/extensions/activity-stream
+    # Copy over the entire repo
+    copy: cpx "./**/{,.}*" $npm_package_config_mc_dir/browser/extensions/activity-stream
```
Comment (Mardak) [inline yamscripts.yml]: This seems wrong to copy over node_modules (even if ignored/not-checked-in-to-m-c) and other activity-stream "built" files/directories that are not used as part of the export. I.e., the mirror should be like a clean checkout of the activity-stream repository with maybe the desired built files.

Comment (Mardak) [reply yamscripts.yml]: In particular, it's trying to copy over 200MB+ of node_modules files to ignore (or worse(?) -- overwrite)

File: yamscripts.yml [hunk 2]
```diff
@@ -21,11 +21,10 @@ scripts:
...
+    version: node ./bin/update-version.js $npm_package_config_mc_dir/browser/extensions/activity-stream/system-addon
     copyPingCentre: cpx "system-addon/ping-centre/PingCentre.jsm" $npm_package_config_mc_dir/browser/modules
-    rmextra: rimraf "$npm_package_config_mc_dir/browser/extensions/activity-stream/{{,test/unit/}{content-src,ping-centre},data/locales.json}"
```
Comment (Mardak) [inline yamscripts.yml]: Some of the things being removed here are the intermediate "built" files, e.g., locales.json, that currently aren't checked in to a-s or m-c.

File: package.json [hunk 3]
```diff
@@ -83,19 +84,18 @@
...
+    "cleanmc": "rimraf $npm_package_config_mc_dir/browser/extensions/activity-stream/system-addon",
     "buildmc": "npm-run-all buildmc:*",
     "prebuildmc": "npm run cleanmc",
     "buildmc:locales": "pontoon-to-json --src locales --dest system-addon/data",
     "buildmc:webpack": "webpack --config webpack.system-addon.config.js --display-optimization-bailout",
     "buildmc:css": "node-sass --source-map true --source-map-contents system-addon/content-src/styles -o system-addon/css",
     "buildmc:html": "rimraf system-addon/prerendered && webpack --config webpack.prerender.config.js && node ./bin/render-activity-stream-html.js",
-    "buildmc:copy": "cpx \"system-addon/**/{,.}*\" $npm_package_config_mc_dir/browser/extensions/activity-stream",
-    "buildmc:version": "node ./bin/update-version.js $npm_package_config_mc_dir/browser/extensions/activity-stream",
+    "buildmc:copy": "node ./bin/mirror-trees.js",
```
Comment (Mardak) [inline package.json]: `cpx` is just using glob, which you're separately depending on now. You should be able to exclude directories with:
`cpx './{{,.}*,!(logs|node_modules)/**}' $npm_package_config_mc_dir/…`

**Comment by k88hudson** [comment]
Closing this in favour of #4169

**Pull request closed without merging**